python/nav/web/utils.py

```diff
@@ -64,3 +71,59 @@ def wrapper(request, *args, **kwargs):
         return wrapper
 
     return wrap
+
+
+def generate_qr_code(url: str, name: str = "") -> io.BytesIO:
+    """
+    Generate a QR code from a given url, and, if given, adds the name below as a
+    caption
+
+    Returns the generated image as a bytes buffer
+    """
+    # Creating QR code
+    qr = qrcode.QRCode(box_size=10)
+    qr.add_data(url)
+    img = qr.make_image()
+    draw = ImageDraw.Draw(img)
+
+    # Adding the name as caption
+    if name:
+        img_width, img_height = img.size
+        font_path = os.path.join(os.path.dirname(__file__), "static/fonts/OS600.woff")
+        if len(name) < 25:
+            font = ImageFont.truetype(font_path, 25)
+        elif len(name) < 50:
+            font = ImageFont.truetype(font_path, 15)
+        else:
+            font = ImageFont.truetype(font_path, 10)
+        caption_width = font.getlength(name)
+        draw.text(
+            ((img_width - caption_width) / 2, img_height - 40),
+            text=name,
+            font=font,
+            fill="black",
+        )
+
+    file_object = io.BytesIO()
+    img.save(file_object, "PNG")
```

Comment:
We might want to consider an option to produce SVG images, if possible from the qrcode library - in which case we wouldn't need to bother with deciding the physical dimensions (and therefore, storage requirements) of the resulting graphic.

But this could be a different PR/issue altogheter.

Reply:
Yes, it is possible to do that with the qrcode library.

I can create a separate issue for that.